### PR TITLE
feat: support azure account name storage options in sync db.connect

### PR DESCRIPTION
**Pull request by LuQQiu**
db.connect with azure storage account name is supported in async connect but not sync connect. 
Add this functionality

### CODE REVIEW

File: python/python/lancedb/remote/db.py [hunk 1]
```diff
@@ -44,8 +44,17 @@ def __init__(
         client_config: Union[ClientConfig, Dict[str, Any], None] = None,
         connection_timeout: Optional[float] = None,
         read_timeout: Optional[float] = None,
+        storage_options: Optional[Dict[str, str]] = None,
     ):
         """Connect to a remote LanceDB database."""
+        if storage_options:
```
Comment (eddyxu) [inline python/python/lancedb/remote/db.py:50]: Can we check this on the server side?

**Comment by LuQQiu** [comment]
@eddyxu @wjones127 Addressed the comment, PTAL

**Comment by QianZhu** [comment]
@LuQQiu Do we plan to add `storage_options` to our nodejs SDK as well? Don't have to address it in this PR

**Comment by LuQQiu** [comment]
@QianZhu will double check with them about their SDK, but yeah, should be supported in both
i will also check the Nodejs SDK part, likely not supported yet

**Review by eddyxu — APPROVED** [review]

**Pull request merged**